tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html

```diff
@@ -3784,8 +3784,52 @@ <h2>Show similarity to selected datapoint</h2>
           }
         },
 
+        computeClosestCounterfactual: function(exInd, distances) {
+          // Distances are indexed by example ids
+          const modelInferenceValueStr = this.strWithModelName_(
+            inferenceValueStr,
+            this.nearestCounterfactualModelIndex
+          );
+          let closestDist = Number.POSITIVE_INFINITY;
+          let closest = -1;
+          for (let i = 0; i < this.visdata.length; i++) {
+            // Skip examples with the same inference class as the selected
+            // examples.
+            if (
+              this.visdata[exInd][modelInferenceValueStr] ==
+              this.visdata[i][modelInferenceValueStr]
+            ) {
+              continue;
+            }
+            let dist = distances[i];
+            if (dist < closestDist) {
+              closestDist = dist;
+              closest = i;
+            }
+          }
+          if (closest != -1) {
+            // Display the counterfactual in dive and example viewer.
+            this.comparedIndices = [closest];
+            this.counterfactualExampleAndInference = this.examplesAndInferences[
+              closest
+            ];
+            this.compareTitle = 'Counterfactual value(s)';
+          }
+        },
+
         findClosestCounterfactual_: function() {
           const selected = this.selected[0];
+          // Custom distance function can only be used when local
+          if (this.local && this.customDistanceFunctionSet) {
+            this.requestDistanceWithCallback(
+              selected,
+              'computeClosestCounterfactual',
+              {callbackParams: {}, distanceParams: {}}
+            );
+            return;
+          }
+
+          // If not using custom distance function compute counterfactuals here.
```

Comment:
dont see the computation that the comment refers to

Reply:
I meant the whole block, could do something nicer in future to avoid code duplication. Moved the comment up.